src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -359,4 +357,59 @@ private ScopeTable newChild() {
       return new ScopeTable(this);
     }
   }
+
+  private static class Scanner {
```

Comment:
A brief function comment would be useful, saying what characters we accept, and that `scan()` returns `EOF`, `IDENT`, or a character, and that in the case of `IDENT`, the token can be fetched via `.token`

Reply:
Done